### PR TITLE
 [ISSUE  #2860]  support opentracing for rocketmq client

### PR DESCRIPTION
**Pull request by yuz10**
**Make sure set the target branch to `develop`**

## [ISSUE  #2860] support opentracing for rocketmq client

support opentracing for rocketmq client

Follow this checklist to help us incorporate your contribution quickly and easily. Notice, `it would be helpful if you could finish the following 5 checklist(the last one is not necessary)before request the community to review your PR`.

- [ :heavy_check_mark:] Make sure there is a [Github issue](https://github.com/apache/rocketmq/issues) filed for the change (usually before you start working on it). Trivial changes like typos do not require a Github issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue. 
- [ :heavy_check_mark:] Format the pull request title like `[ISSUE #123] Fix UnknownException when host config not exist`. Each commit in the pull request should have a meaningful subject line and body.
- [ :heavy_check_mark:] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [ :heavy_check_mark:] Write necessary unit-test(over 80% coverage) to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add integration-test in [test module](https://github.com/apache/rocketmq/tree/master/test).
- [ :heavy_check_mark:] Run `mvn -B clean apache-rat:check findbugs:findbugs checkstyle:checkstyle` to make sure basic checks pass. Run `mvn clean install -DskipITs` to make sure unit-test pass. Run `mvn clean test-compile failsafe:integration-test`  to make sure integration-test pass.
- [ x ] If this contribution is large, please file an [Apache Individual Contributor License Agreement](http://www.apache.org/licenses/#clas).

### CODE REVIEW

File: client/pom.xml [hunk 1]
```diff
@@ -47,6 +47,18 @@
             <groupId>org.apache.commons</groupId>
             <artifactId>commons-lang3</artifactId>
         </dependency>
+        <dependency>
+            <groupId>io.opentracing</groupId>
+            <artifactId>opentracing-api</artifactId>
+            <version>0.33.0</version>
+            <scope>provided</scope>
```
Comment (vongosling) [inline client/pom.xml:54]: Good catch

File: example/pom.xml [hunk 2]
```diff
@@ -52,5 +52,15 @@
             <groupId>org.javassist</groupId>
             <artifactId>javassist</artifactId>
         </dependency>
+        <dependency>
+            <groupId>io.jaegertracing</groupId>
+            <artifactId>jaeger-core</artifactId>
+            <version>1.6.0</version>
```
[the remark this answers is not in the document]
Comment (wu-sheng) [reply example/pom.xml:58]: Hi, thanks for reaching me here. SkyWalking supports RocketMQ through our plugin system, so, we don't need to rely on the OpenTracing APIs layer, also don't ask the RocketMQ community to change the source code.

Also, as a member of OpenTracing, I have to say, OpenTracing is no longer active anymore. The members are either quit or join OpenTelemetry. OpenTracing and OpenCensus have been merged. SkyWalking community already doesn't recommend users to choose OpenTracing-SkyWalking bridge.

RocketMQ 3.x and 4.x plugins are available here
1. https://github.com/apache/skywalking/tree/master/apm-sniffer/apm-sdk-plugin/rocketMQ-4.x-plugin
2. https://github.com/apache/skywalking/tree/master/apm-sniffer/apm-sdk-plugin/rocketMQ-3.x-plugin

[the remark this answers is not in the document]
Comment (vongosling) [reply example/pom.xml:58]: Very helpful. OpenTelemetry is a very promising direction. Until it's fully integrated, the community will prefer two more open approaches :-)

**Review by vongosling — APPROVED** [review]

File: example/pom.xml (hunk 2 above)
[the remark this answers is not in the document]
Comment (wu-sheng) [reply example/pom.xml:58]: If people really ask about OpenTracing + SkyWalking, I am thinking no one will, just tell them use our agent. All things are going to be out of the box.

[the remark this answers is not in the document]
Comment (wu-sheng) [reply example/pom.xml:58]: The good thing is, SkyWalking has a version identification feature, so, it only works if your user is still there. And no harm if there isn't.

[the remark this answers is not in the document]
Comment (wu-sheng) [reply example/pom.xml:58]: @yuz10 I would recommend you take a look how SkyWalking's OT bridge works, you may change your mind after you read codes :) We are not working like Jaeger.

[the remark this answers is not in the document]
Comment (wu-sheng) [reply example/pom.xml:58]: And OT community is bad of back-compatible, so 0.33 is not working with skywalking OT bridge. It was 0.31 if I recommend it correctly. And the crazy thing is, 0.33 and 0.31 have a big gap. Who knows.

[the remark this answers is not in the document]
Comment (vongosling) [reply example/pom.xml:58]: Nice reminder~ nowadays, its dependency is provided scope. I would like to keep a close eye on this field.

**Pull request merged**